pkg/healthcheck/healthcheck.go

```diff
@@ -1594,6 +1600,31 @@ func (hc *HealthChecker) RunChecks(observer CheckObserver) (bool, bool) {
 	return success, warning
 }
 
+func (hc *HealthChecker) RunChecksExitOnError() (bool, bool) {
+	return hc.RunChecks(func(result *CheckResult) {
+		if result.Retry {
+			fmt.Fprintln(os.Stderr, "Waiting for control plane to become available")
+			return
+		}
+
+		if result.Err != nil && !result.Warning {
+			var msg string
+			switch result.Category {
+			case KubernetesAPIChecks:
+				msg = "Cannot connect to Kubernetes"
+			case LinkerdControlPlaneExistenceChecks:
+				msg = "Cannot find Linkerd"
+			}
+			fmt.Fprintf(os.Stderr, "%s: %s\n", msg, result.Err)
+
+			checkCmd := "linkerd check"
+			fmt.Fprintf(os.Stderr, "Validate the install with: %s\n", checkCmd)
```

Comment:
If we are using this in the `viz` extension, do we expect the `checkCmd` to be `linkerd viz check`? Similar question for all other extensions. If we don't, we can skip the assignment and go straight to `fmt.Fprintln(os.Stderr, "Validate the install with 'linkerd check'")`